linux/keyman-config/keyman_config/list_installed_kmp.py

```diff
@@ -122,32 +122,21 @@ def get_kmp_version(packageID):
         None: if not found
     """
     version = None
-    user_kmp = get_installed_kmp(InstallLocation.User)
-    shared_kmp = get_installed_kmp(InstallLocation.Shared)
-    os_kmp = get_installed_kmp(InstallLocation.OS)
-
-    if packageID in os_kmp:
-        version = os_kmp[packageID]['version']
-
-    if packageID in shared_kmp:
-        shared_version = shared_kmp[packageID]['version']
-        if version:
-            if version < shared_version:
-                version = shared_version
-        else:
-            version = shared_version
-
-    if packageID in user_kmp:
-        user_version = user_kmp[packageID]['version']
-        if version:
-            if version < user_version:
-                version = user_version
-        else:
-            version = user_version
-
+    version = _get_kmp_version_internal(packageID, InstallLocation.OS, version)
+    version = _get_kmp_version_internal(packageID, InstallLocation.Shared, version)
+    version = _get_kmp_version_internal(packageID, InstallLocation.User, version)
     return version
 
 
+def _get_kmp_version_internal(packageId, location, previousVersion):
+    kmp = get_installed_kmp(location)
+    if packageId in kmp:
+        version = kmp[packageId]['kmpversion']
+        if previousVersion and previousVersion < version or not previousVersion:
```

Comment:
String comparison of version numbers is not safe: `'2.0.1' > '10.0'`.

Also, this if statement is kinda strange, mixing `and` and `or not` like that, perhaps could do instead (still need to fix the version comparison though):

```suggestion
        if not previousVersion or previousVersion < version:
```

Reply:
We changed the version number comparison a while ago, but looks this one slipped through...